src/main/java/seedu/todo/ui/InputHandler.java

```diff
@@ -1,12 +1,95 @@
 package seedu.todo.ui;
 
+import java.util.LinkedList;
+
 import seedu.todo.controllers.*;
 
 public class InputHandler {
+    
+    private static InputHandler instance;
+    
+    private static final int MAX_HISTORY_SIZE = 20;
+    private static final int INITIAL_HISTORY_INDEX = -1;
+    private static LinkedList<String> commandHistory = new LinkedList<String>();
```

Comment:
Okay, but I see it as minimal improvement, seeing that using ListIterator and `.next` and `.previous` are already O(1), no? The advantages are purely at the caching level. Besides, I already done with listIterator lol